assets/javascripts/modules/checkout/promoCode.js

```diff
@@ -22,24 +23,79 @@ define([
         bindExtraKeyListener.alreadyBound = true;
     }
 
-    function displayError(message) {
-        message = message || 'Invalid promo code, please try again.';
+    function unapplyPromotionToRatePlans() {
+        $ratePlanFields.each(function(el) {
+            var $el = $(el),
+                currency = $el.data('currency'),
+                $label = $('#label-for-' + $el.val() + '-' + currency),
+                newDisplayPrice = $el.data('option-mirror-label-default');
+
+            $label.html(newDisplayPrice);
+            $el.attr('data-option-mirror-label', newDisplayPrice);
+            if ($el.attr('checked')) {
+                bean.fire(el, 'change');
+            }
+        });
+    }
+
+    function formatMoney(currencyIdentifier, amount) {
+        // rounds any remainder up tp 1p so we always quote a higher price than the user might eventually get charged.
+        var price2dp = Math.ceil(amount * 100) / 100,
+            priceNoDot00 = price2dp.toFixed(2).replace(/\.00$/, '');
+
+        return currencyIdentifier + priceNoDot00;
+    }
+
+    function applyPromotionToRatePlans(promotion) {
+        if (promotion.promotionType.amount > 0) {
+            $ratePlanFields.each(function(el) {
+                var $el = $(el),
+                    amount = Number($el.data('amount-in-currency')),
+                    currency = $el.data('currency'),
+                    currencyIdentifier = $el.data('currency-identifier'),
+                    discountPercent = promotion.promotionType.amount,
+                    frequency = $el.data('frequency'),
+                    $label = $('#label-for-' + $el.val() + '-' + currency),
+                    newDisplayPrice = formatMoney(currencyIdentifier, amount * (1 - (discountPercent / 100))) + ' ' + frequency;
+
+                $label.html(newDisplayPrice);
+                $el.attr('data-option-mirror-label', newDisplayPrice);
```

Comment:
using $el.data('option-mirror-label') wouldn't work as $.data() does not set back in to the DOM, and so wouldn't get picked up by the mirrorLabel.js module which reads the attribute.